src/main.cpp

```diff
@@ -32,8 +49,19 @@ int main() {
 			return;
 		}
 
+		std::ifstream guild_config("../" + event.msg.guild_id.str() + ".json");
+
+		if (guild_config.bad()) {
+			event.reply("Could not give a role based off this message as the role has not been setup! Please use `/set_role`.");
+			return;
+		}
+
+		json j;
+		guild_config >> j;
+
 		const dpp::message& msg = event.msg;
 		const dpp::user& user = event.msg.interaction.usr;
+		const dpp::snowflake role_id{j["ROLE_ID"]};
```

Comment:
`=` and `{}` do different things here. The other two are `=` because `&`. `role_id` has to be created, so we use `{}`.

For variable initialisation, you should almost always do `{}` rather than `=` or `()`